File: src/test/java/org/neo/gomina/model/scm/impl/TmateSoftSvnClientTest.java
package org.neo.gomina.model.scm.impl;

import org.junit.Test;
import org.neo.gomina.model.scm.model.Commit;

import java.util.List;

public class TmateSoftSvnClientTest {


    @Test
    public void testSvn() throws Exception {
        TmateSoftSvnClient svnClient = new TmateSoftSvnClient("file:////Users/Amokrane/Work/SvnRepo/svn-repo-demo");

        List<Commit> svnLog = svnClient.getLog("svn-project2", "0", 100);
        for (Commit commit : svnLog) {
            System.out.println(commit);
        }
    }

}

File: src/test/java/org/neo/gomina/integration/scm/svn/TmateSoftSvnClientTest.kt
package org.neo.gomina.integration.scm.svn

import org.fest.assertions.Assertions
import org.junit.Test

class TmateSoftSvnClientTest {

    val root = "file:////Users/Shared/SvnRepo/svn-repo-demo/"
    val projectUrl = "svn-project1"

    @Test
    fun testSvn() {
        val svnClient = TmateSoftSvnClient(root, projectUrl)

        val svnLog = svnClient.getLog("trunk", "0", 100)
        svnLog.forEach { println(it) }
        Assertions.assertThat(svnLog.size).isGreaterThan(16)
    }

    @Test
    fun testListFiles() {
        val svnClient = TmateSoftSvnClient(root, projectUrl)
        svnClient.listFiles("/", "-1").forEach { println(it) }
    }

    @Test
    fun testGetFile() {
        val svnClient = TmateSoftSvnClient(root, projectUrl)
        svnClient.getFile("trunk", "README.md", "-1")?.let { println(it) }
    }

    @Test
    fun testBranches() {
        val svnClient = TmateSoftSvnClient(root, projectUrl)
        svnClient.getBranches().forEach { println("branch: $it") }
    }
}